src/nitllvm.nit

```diff
@@ -0,0 +1,155 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# LLVM backend for the Nit compiler (Work in progress)
+#
+# # Features and TODO list
+#
+# - [x] Generate basic class structures
+# - [x] Generate LLVM function signatures
+# - [ ] Generate LLVM implementation code
+# - [ ] Attributes
+# - [ ] Intern methods
+# - [ ] Generic and other types
+# - [ ] Support the FFI with C
+#
+# # References
+#
+# * http://llvm.org/docs/WritingAnLLVMBackend.html
+# * http://llvm.org/docs/LangRef.html
+# * http://llvm.org/releases/2.6/docs/tutorial/JITTutorial1.html
+module nitllvm
+
+import modelbuilder
+import frontend
+import model_utils
+
+redef class ToolContext
+	# --output
+	var opt_output = new OptionString("Output file", "-o", "--output")
+
+	init do option_context.add_option opt_output
+
+	# LLVM code generation
+	var llvm_phase: Phase = new LLVMPhase(self, null)
+end
+
+private class LLVMPhase
+	super Phase
+
+	redef fun process_mainmodule(mainmodule, mmodules)
+	do
+		# Prepare LLVM code in memory
+		var llvm_classes_vfts = new HashMap[String, String]
+		var llvm_functions = new HashMap[String, String]
+		var llvm_attributes = new HashMap[String, String]
+
+		for mclass in mainmodule.flatten_mclass_hierarchy do
+			var class_vft = ["%class_{mclass.name}_vft = type \{\n"]
+			var class_vft_entries = new Array[String]
+			var class_attributes = ["\n%class_{mclass.name} = type \{%class_{mclass.name}_vft*"] # TODO
+
+			var mproperties = mclass.all_mproperties(mainmodule, private_visibility)
+			for mproperty in mproperties do
```

Comment:
The order of `mproperties` does not offer any guarantee. Especially no guarantee about any position invariance with regard to super-classes.